src/github.com/matrix-org/dendrite/clientapi/auth/storage/devices/devices_table.go

```diff
@@ -140,6 +153,44 @@ func (s *devicesStatements) selectDeviceByToken(
 	return &dev, err
 }
 
+func (s *devicesStatements) selectDeviceByID(
+	ctx context.Context, id string,
```

Comment:
I think device IDs are scoped to a user, so this should probably take a localpart as an argument.

I'd also rename `id` to `device_id` just to make it a bit more obvious which sort of ID you're talking about